include/zephyr/pm/policy.h

```diff
@@ -275,63 +267,140 @@ static inline int32_t pm_policy_next_event_ticks(void)
  * The system will not enter any power state that would make the system to
  * exceed the given latency value.
  *
+ * If immediate control manager is provided request may trigger an action asynchronous action
+ * to reconfigure the system runtime to fullfil requested latency. @p req contains an
+ * asynchronous notification field that must be initialize prior to the call to get the
+ * notification.
+ *
  * @param req Latency request.
  * @param value_us Maximum allowed latency in microseconds.
+ *
+ * @retval 0 if request is applied.
+ * @retval 1 if request required immediate action that is not completed. Configured completion
+ * notification will inform about completion.
```

Comment:
In async functions, I would expect that the callback is called regardless of the returned value. So this could be simplified to `retval 0 if the request is pending. When the callback is applied, the callback notification is called with the result`.